File: solutions/cpp/1169.cpp
class Solution {
 public:
  vector<string> invalidTransactions(vector<string>& transactions) {
    unordered_map<string, vector<vector<string>>> map;
    unordered_set<string> ans;

    for (const string& transaction : transactions) {
      istringstream ss(transaction);
      vector<string> token(4, "");
      for (int i = 0; getline(ss, token[i++], ',');)
        ;
      if (stoi(token[2]) > 1000) ans.insert(transaction);
      for (const auto& val : map[token[0]])
        if (abs(stoi(val[1]) - stoi(token[1])) <= 60 && val[3] != token[3]) {
          ans.insert(transaction);
          ans.insert(val[0] + "," + val[1] + "," + val[2] + "," + val[3]);
        }
      map[token[0]].push_back({token[0], token[1], token[2], token[3]});
    }

    return {begin(ans), end(ans)};
  }
};

File: solutions/java/1169.java
class Solution {
  public List<String> invalidTransactions(String[] transactions) {
    Map<String, List<String[]>> map = new HashMap<>();
    Set<String> ans = new HashSet<>();

    for (String transaction : transactions) {
      String[] token = transaction.split(",");
      if (Integer.valueOf(token[2]) > 1000)
        ans.add(transaction);
      if (map.containsKey(token[0]))
        for (String[] val : map.get(token[0]))
          if (Math.abs(Integer.valueOf(val[1]) - Integer.valueOf(token[1])) <= 60 && !val[3].equals(token[3])) {
            ans.add(transaction);
            ans.add(val[0] + "," + val[1] + "," + val[2] + "," + val[3]);
          }
      List<String[]> list = map.getOrDefault(token[0], new ArrayList<>());
      list.add(new String[] { token[0], token[1], token[2], token[3] });
      map.put(token[0], list);
    }

    return new ArrayList<>(ans);
  }
}